server/core/src/main/java/org/infinispan/server/core/configuration/SslConfiguration.java

```diff
@@ -18,7 +18,7 @@
    private final boolean requireClientAuth;
    private final Map<String, SslEngineConfiguration> sniDomainsConfiguration;
 
-   SslConfiguration(boolean enabled, boolean requireClientAuth, Map<String, SslEngineConfiguration> sniDomainsConfiguration) {
+   public SslConfiguration(boolean enabled, boolean requireClientAuth, Map<String, SslEngineConfiguration> sniDomainsConfiguration) {
```

Comment:
Why does this need to be public? It doesn't appear to be used anywhere other than in `SslConfigurationBuilder`

Reply:
It's used as a part of a Single Port configuration. It turned out that `NettyTransport` takes `SslConfiguration` and `SslEngineConfiguration` and prepares `SslContext` based on that. So the easiest way to reuse `NettyTransport` is to feed it with a `SslConfiguration` instance.

Unfortunately using a `SslConfigurationBuilder` is a bit hard since it accepts `ProtocolServerConfigurationChildBuilder` in its constructor (and I don't have it at hand since a Single Port has different configuration root).

So here are the options I've been thinking about to solve this puzzle:
* Make `SslConfiguration` public [not very nice but short and simple].
* Add new constructor without `ProtocolServerConfigurationChildBuilder` to `SslConfigurationBuilder` [should work fine but you might get `NullPointerException`s on some method calls, which feels a bit wrong].
* Create interfaces for `SslConfiguration` and `SslEngineConfiguration` and implement them in Single Port Router [lots and lots of typing and duplicating code].
* Forget about `NettyTransport` and bootstrap Netty manually in a Single Port module [lots of duplicated code handling SSL and ALPN].

So I decided to take the shortest route even though it's a bit ugly....

Reply:
But I'm absolutely open to some new options how to wire those bits up...

Reply:
At the minute your code utilises `SslConfigurationBuilder(null)` in the `SinglePortRouterBuilder` and `new SslConfiguration(..)` is never called outside of the builder. So this is a bit different to your description unless I am missing something. Do you intend to use the `SslConfiguration` constructor later?

Reply:
But I appreciate all of your points and can't think of a cleaner way, so if making SslConfiguration's constructor public is the least painful way, then so be it :+1:

Reply:
Done